### ISSUE TITLE
Review accessibility of native `<details>` element

### ISSUE THREAD

**Issue by romaricpascal** [alphagov/govuk-frontend#4474]
## What

Go through the browser/assistive tech combinations of our accessibility test spreadsheet and verify that the details element behaves as intended. Especially [NVDA with Chrome and VoiceOver on iOS](https://cloudfour.com/thinks/a-details-element-as-a-burger-menu-is-not-accessible/#the-expanded-collapsed-state-is-not-conveyed).

## Why

We've removed the JavaScript polyfill for the Details element. As this is quite a change, we need to check accessibility is still alright.

## Who needs to work on this

Developers

## Who needs to review this

Developers

## Done when

- [ ] We're reviewed the accessibility of the native `<details>` element and if necessary planned any fix.

**Comment by domoscargin** [comment]
I've had a gander at this and I suspect no further testing is necessary for the scope of this issue.

The original JavaScript started with a [check to see if the browser supported native `<details>`](https://github.com/alphagov/govuk-frontend/blob/fbd82b327a4203dae8ac5871eb9dbacee8936911/packages/govuk-frontend/src/govuk/components/details/details.mjs#L44-L46). If it did, we just used the native `<details>` with our styles. We only polyfilled if the browser didn't already support it natively.

So nothing has actually changed in that regard. There were some other accessibility issues flagged for this component from yonks ago, but it looks like those are getting fixed by vendors: https://github.com/alphagov/govuk-frontend/issues/2349#issuecomment-1816266686

Suggest we close this.

**Comment by 36degrees** [comment]
I agree with your conclusions 👍🏻 There should be no material change in browsers that natively support the `<details>` element, and in other browsers it now falls back to text and is therefore non-interactive.
